src/components/__tests__/App-test.tsx

```diff
@@ -0,0 +1,25 @@
+import { expect } from "chai";
+
+import * as React from "react";
+import * as Enzyme from "enzyme";
+import { shallow } from "enzyme";
+import App from "../App";
+import Header from "../Header";
+import LibrariesListContainer from "../LibrariesListContainer";
+
+describe("App", () => {
+  let wrapper: Enzyme.ShallowWrapper<any, {}>;
+  beforeEach(() => {
+    wrapper = shallow(
+      <App />
+    );
+  });
+  it("should render the header", () => {
+    let header = wrapper.find(Header);
+    expect(header.length).to.equal(1);
+  });
+  it("should render the libraries list container", () => {
+    let container = wrapper.find(LibrariesListContainer);
+    expect(container.length).to.equal(1);
```

Comment:
Would you expect this to have a length one if no data is passed to it on a shallow mount?

Reply:
The container would still be there--it would just be empty...

Reply:
Is that a good test though?

Reply:
Or is that what a React component should return when no data is passed to it?

Reply:
I think the point is that the top-level component for the whole app should still render the same basic architecture irrespective of whether there happens to be data.  Initializing the app with no data, e.g. if there are no libraries in this registry, should always give you a container (it doesn't matter whether the container is empty or full; the container doesn't even know anything about that)--it shouldn't give you a blank screen or an error message or anything like that.

Reply:
A fallback component (or error message) to render when there is no data to display is a better approach instead of displaying nothing. Then the user would be aware of what is going on instead of getting no feedback. Granted, this app won't have a big user base, but it's good to get into the mindset of building more progressive apps regardless of who uses it for the sake of good implementation habits.